### PR TITLE
Data model supports multiple providers and default provider

### PR DESCRIPTION
**Pull request by mehansen**
# BACKEND PULL REQUEST

## Related Issue

first of probably 5 PRs to implement #7982 

## Changes Proposed

- add concept of multiple providers to a facility
- add default ordering provider column, populate it with ordering provider data, and swap all references to ordering provider to instead get default provider or a provider from the facility's provider set
- when a new facility is created, the ordering provider is set as the default and added to the set of providers

## Additional Information
next PR will remove the ordering provider column (doing it in a separate merge to make rollbacks less risky)

next next PR will handle:
- backend will support adding, removing, listing providers and getting/updating the default provider
- update org initialization logic to fix how we're creating the default provider and facilities (e.g. see number of identical providers in your local DB)

## Testing

DB migration should populate `facility_providers` table and new `default_ordering_provider_id` column on `facility` table

test that existing provider related behavior is not impacted:
- create facility with a provider
- update provider
- delete a facility (provider is not marked as deleted)

### CODE REVIEW

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 1]
```diff
@@ -5700,4 +5696,93 @@ databaseChangeLog:
...
+                    foreignKeyName: fk__facility_providers__facility
+                    references: facility
+              - column:
+                  name: provider_id
+                  type: uuid
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk__facility_providers__provider
+                    references: provider
+  - changeSet:
```
Comment (emyl3) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:5723]: this change is just so we can view the new table from metabase right?

Comment (mehansen) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:5723]: yes that's right!

Comment (mehansen) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:5723]: oh uh oh it doesn't seem to have worked though now that I'm looking at dev4 metabase. investigating...

**Review by bobbywells52 — APPROVED** [review]
LGTM -- thanks for your work on this!

File: backend/src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -5718,4 +5713,94 @@ databaseChangeLog:
                    (gen_random_uuid(), 'Chlamydia', 'LP14298-1')
       rollback:
         - sql:
-            sql: DELETE FROM ${database.defaultSchemaName}.supported_disease WHERE name = 'Chlamydia';
\ No newline at end of file
+            sql: DELETE FROM ${database.defaultSchemaName}.supported_disease WHERE name = 'Chlamydia';
+  - changeSet:
```
Comment (emyl3) [inline backend/src/main/resources/db/changelog/db.changelog-master.yaml:5717]: do we want a rollback on this to drop this table?

Comment (mehansen) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:5717]: createTable (also the addColumn changeset) can be [auto-rolled back](https://docs.liquibase.com/workflows/liquibase-community/automatic-custom-rollbacks.html) so we don't have to write custom rollback statements for them

Comment (emyl3) [reply backend/src/main/resources/db/changelog/db.changelog-master.yaml:5717]: ooo right 🤦🏻 thank you for the reminder

**Review by mpbrown — APPROVED** [review]
LGTM! Nice work on this Merethe!

**Pull request merged**